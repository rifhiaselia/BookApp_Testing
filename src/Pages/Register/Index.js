import {
  StyleSheet,
  Text,
  View,
  TextInput,
  Button,
  TouchableOpacity,
  Image,
  Alert,
} from 'react-native';
import React, {useState} from 'react';
import {Back} from '../../Assets/Icons';
import {useDispatch, useSelector} from 'react-redux';
import {RegisterSuccess} from '../../actions';

const Register = ({navigation}) => {
  const dispatch = useDispatch();

  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const isRegister = useSelector(state => {
    return state.appData.isRegister;
  });

  function kirimData() {
    const dataObject = {
      name: name,
      email: email,
      password: password,
    };
    console.log('form', dataObject);
    dispatch(RegisterSuccess(dataObject, navigation));
    // isRegister ? navigation.navigate('Success') : Alert.alert('Register Gagal');
  }

  return (
    <View style={styles.Container}>
      <TouchableOpacity onPress={() => navigation.goBack()}>
        <Image style={styles.back} source={Back} />
      </TouchableOpacity>
      <Text style={styles.text1}>Create Account</Text>
      <View style={{marginTop: 10, marginBottom: 10}}>
        <TextInput
          testID="input-name"
          style={styles.text}
          placeholder="Full Name"
          value={name}
          onChangeText={text => setName(text)}
        />
        {/* <TextInput
          testID="input-email"
          style={styles.text}
          placeholder="Email"
          value={email}
          onChangeText={text => setEmail(text)}
        /> */}
        <TextInput
          testID="input-email"
          style={styles.text}
          placeholder="Email"
          value={email}
          onChangeText={text => setEmail(text)}
          secureTextEntry={true}
        />
        <TextInput
          testID="input-password"
          style={styles.text}
          placeholder="Password"
          value={password}
          onChangeText={text => setPassword(text)}
          secureTextEntry={true}
        />
      </View>
      <Button testID="submitregister" title="REGISTER" onPress={kirimData} />
      <Text style={styles.text2}>Log into existing account</Text>
      <Button title="LOGIN" onPress={() => navigation.navigate('Login')} />
    </View>
  );
};

export default Register;

const styles = StyleSheet.create({
  back: {
    height: 40,
    width: 40,
    marginBottom: 10,
  },
  Container: {
    flex: 1,
    padding: 20,
    backgroundColor: '#ffffff',
  },
  text: {
    borderWidth: 1,
    marginBottom: 10,
    borderColor: 'grey',
    padding: 10,
    borderRadius: 5,
    backgroundColor: '#DEE6EF',
  },
  text1: {
    fontSize: 30,
    marginBottom: 10,
    color: '#05070d',
    fontWeight: 'bold',
    padding: 10,
    marginTop: 15,
  },
  text2: {
    textAlign: 'center',
    marginVertical: 10,
    fontSize: 15,
    color: '#05070d',
  },
});
