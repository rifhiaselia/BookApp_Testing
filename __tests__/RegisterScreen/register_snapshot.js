import React from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {shallow, configure} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import renderer from 'react-test-renderer';
import Register from '../../src/Pages/Register';

configure({adapter: new Adapter(), disableLifecycleMethods: true});
const homeWrapper = shallow(<Register />);

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch,
}));
jest.mock('react-native-push-notification', () => ({
  addEventListener: jest.fn(),
  requestPermissions: jest.fn(),
  then: jest.fn(),
}));
jest.mock('react-native-share', () => ({}));
jest.mock('react-native-video', () => 'Video');
jest.mock('react-native-pdf', () => 'Pdf');
const mockDispatch = jest.fn();

describe('test snapshot register', () => {
  it('test berhasil', async () => {
    const snapshot = renderer.create(<Register />).toJSON();
    expect(snapshot).toMatchSnapshot()
  });
//   it('test berhasil', async () => {
//     expect(snapshot).toMatchSnapshot()
//   });
});
