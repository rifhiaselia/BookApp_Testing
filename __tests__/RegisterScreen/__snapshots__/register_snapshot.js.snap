// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot register test berhasil 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/Assets/Icons/Back.png",
        }
      }
      style={
        Object {
          "height": 40,
          "marginBottom": 10,
          "width": 40,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#05070d",
        "fontSize": 30,
        "fontWeight": "bold",
        "marginBottom": 10,
        "marginTop": 15,
        "padding": 10,
      }
    }
  >
    Create Account
  </Text>
  <View
    style={
      Object {
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Full Name"
      style={
        Object {
          "backgroundColor": "#DEE6EF",
          "borderColor": "grey",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginBottom": 10,
          "padding": 10,
        }
      }
      testID="input-name"
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#DEE6EF",
          "borderColor": "grey",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginBottom": 10,
          "padding": 10,
        }
      }
      testID="input-email"
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "backgroundColor": "#DEE6EF",
          "borderColor": "grey",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginBottom": 10,
          "padding": 10,
        }
      }
      testID="input-password"
      value=""
    />
  </View>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="submitregister"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        REGISTER
      </Text>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#05070d",
        "fontSize": 15,
        "marginVertical": 10,
        "textAlign": "center",
      }
    }
  >
    Log into existing account
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        LOGIN
      </Text>
    </View>
  </View>
</View>
`;
